core/watcher/watchertest/secrets.go

```diff
@@ -5,128 +5,49 @@ package watchertest
 
 import (
 	"sort"
-	"time"
 
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
 	"github.com/juju/juju/core/watcher"
-	"github.com/juju/juju/internal/testing"
+	"github.com/juju/juju/internal/utils"
 )
 
-// SecretsTriggerWatcherC embeds a gocheck.C and adds methods to help
-// verify the behaviour of any watcher that uses a
-// <-chan []SecretTriggerChange
-type SecretsTriggerWatcherC struct {
-	*gc.C
-	Watcher watcher.SecretTriggerWatcher
-}
-
-// NewSecretsTriggerWatcherC returns a SecretsTriggerWatcherC that
-// checks for aggressive event coalescence.
-func NewSecretsTriggerWatcherC(c *gc.C, w watcher.SecretTriggerWatcher) SecretsTriggerWatcherC {
-	return SecretsTriggerWatcherC{
-		C:       c,
-		Watcher: w,
-	}
-}
-
-func (c SecretsTriggerWatcherC) AssertNoChange() {
-	select {
-	case actual, ok := <-c.Watcher.Changes():
-		c.Fatalf("watcher sent unexpected change: (%v, %v)", actual, ok)
-	case <-time.After(testing.ShortWait):
-	}
-}
-
-// AssertChange asserts the given changes was reported by the watcher,
-// but does not assume there are no following changes.
-func (c SecretsTriggerWatcherC) AssertChange(expect ...watcher.SecretTriggerChange) {
-	var received []watcher.SecretTriggerChange
-	timeout := time.After(testing.LongWait)
-	for a := testing.LongAttempt.Start(); a.Next(); {
-		select {
-		case actual, ok := <-c.Watcher.Changes():
-			c.Logf("Secrets Trigger Watcher.Changes() => %# v", actual)
-			c.Assert(ok, jc.IsTrue)
-			received = append(received, actual...)
-			if len(received) >= len(expect) {
-				mc := jc.NewMultiChecker()
-				mc.AddExpr(`_[_].NextTriggerTime`, jc.Almost, jc.ExpectedValue)
-				c.Assert(received, mc, expect)
-				return
-			}
-		case <-timeout:
-			c.Fatalf("watcher did not send change")
+// SecretTriggerTimeAssert is an assert function used as parameter for the
```

Comment:
**suggestion:** I think this comment lose part of information of what it is replacing:

> // AssertChange asserts the given changes was reported by the watcher,
> // but does not assume there are no following changes.

We lose the fact that the expected time range of changes may be shorter than the obtained one. It is an assertion that check we have "at least those change at some given times".